aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -149,6 +163,15 @@ void startDrainingMutationBuffer() {
         );
     }
 
+    /**
+     * Checks if the subscription processor is listening
+     * for events.
+     * @return true if there are listeners. False otherwise.
+     */
+    boolean isObserving() {
+        return disposable.size() > 0;
+    }
+
```

Comment:
This one is tricker than in the other case, because the `SubscriptionProcessor` has an intermediate state, where it is buffering content, but is not processing it. So, we should decided which state we mean. From your current implementation, it would match two states:
- subscribed, not processing
- subscribed, processing under way

```suggestion
    /**
     * Checks if the subscription processor is currently buffering events from the remote AppSync instance.
     * @return true if the subscription processor has active subscriptions to AppSync
     */
    boolean isSubscribed() {
        return !disposable.isEmpty();
    }

```
The other meaning would also be easy to implement:
```
/**
 * Checks if the subscription processor is actively processing events found on its AppSync subscriptions.
 * @return true if the subscription processor is actively processing subscription events
 */
public isDrainingEventBuffer() {
    return disposable.size() == 2;
}
```